samples/apps/bookinfo/src/productpage/Dockerfile.sidecar

```diff
@@ -0,0 +1,37 @@
+# Copyright 2017 Istio Authors
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+FROM python:2-onbuild
+
+ARG PROXY
+
+RUN mkdir -p /etc/istio/proxy/
+RUN apt-get update && apt-get -y install iptables curl
+RUN adduser -disabled-password --gecos "" --uid 1337 istio-proxy
+RUN chown istio-proxy /etc/istio/proxy
+
+RUN echo "download and extract the proxy: https://storage.googleapis.com/istio-build/proxy/envoy-$PROXY.tar.gz"
```

Comment:
thanks it's better than hardcoded but it'd be even better outside and copied 
also there is a fair bit of duplication with https://github.com/istio/proxy/pull/446
(but maybe we can do a grand unification as part of hybrid in 0.3)

Reply:
And maybe download and install the .deb file.

Reply:
yep, would be good to park that for 0.3.  I haven't seen and figure out how to run the .deb file yet.

Reply:
"it'd be even better outside and copied" - @ldemailly could you clarify this?

Reply:
I meant build or download the file before invoking docker build (in a makefile or part of invoking docker); like we do for most other docker images

Reply:
ah, thanks!  This would cut 4 downloads to 1 download since we are building 4 images.

Reply:
addressed thanks!